### ISSUE TITLE
FilePicker returns null when running iOS app on macOS (not Catalyst)

### ISSUE THREAD

**Issue by follesoe** [dotnet/maui#25661]
### Description

When using the `FilePicker` API in an iOS application, the `PickAsync` method returns null (as if canceled) even if the user has not canceled or picked a file.

The same behavior exists in the old Xamarin.Essentials library. On macOS Catalyst, I believe you have worked around the problem by adding in `#if !MACCATALYST` here: https://github.com/dotnet/maui/blob/main/src/Essentials/src/FilePicker/FilePicker.ios.cs#L41 - which is not present in https://github.com/xamarin/Essentials/blob/main/Xamarin.Essentials/FilePicker/FilePicker.ios.cs#L35

However, I believe we are getting the "macOS Catalyst" behavior of the `UIDocumentPickerViewController` when an iOS app runs on an M-series Mac.

One possible solution is using `NSProcessInfo.ProcessInfo.IsiOSApplicationOnMac;` to detect if the iOS app runs on macOS:

```csharp
#if !MACCATALYST
			if (documentPicker.PresentationController != null && !NSProcessInfo.ProcessInfo.IsiOSApplicationOnMac)
			{
				documentPicker.PresentationController.Delegate =
					new UIPresentationControllerDelegate(() => GetFileResults(null, tcs));
			}
#endif
```

### Steps to Reproduce

1. Create a new MAUI application using the FilePicker API
2. Build for iOS and distribute the app through TestFlight (unfortunately MAUI does not support running iOS apps on macOS directly as per: https://github.com/xamarin/xamarin-macios/issues/20868)
3. See that `PickAsync` returns `null` even if file picker is not dismissed.

### Version with bug

9.0.0-rc.2.24503.2

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

**Comment by follesoe** [comment]
The issues reported by the `simular-issues-ai` are all relevant but do not cover this case, as they focus on Mac Catalyst, while I believe the same bug exists when running iOS apps that are not Catalyst on macOS.

**Comment by mattleibow** [comment]
If you run the app, can you see if `OperatingSystem.IsMacCatalyst()` and/or `OperatingSystem.IsMacOS()` returns true?

I wonder if we should switch the directive with a runtime check.

**Comment by follesoe** [comment]
It would be great if this PR/change could get some attention. It's a really minor fix, but the bug is pretty severe when running iOS apps on macOS. cc @mattleibow.

**Comment by follesoe** [comment]
@MartyIX I do not think this will suffice, you need the `!(OperatingSystem.IsIOSVersionAtLeast(14, 0) && NSProcessInfo.ProcessInfo.IsiOSApplicationOnMac)` check to detect iOS running on macOS. In your change set, the code will still be as today (the `if (!OperatingSystem.IsMacCatalyst() && documentPicker.PresentationController is not null)` will return true, and the `documentPicker.PresentationController.Delegate` will be set, which is not what we want.

See this comment: https://github.com/dotnet/maui/pull/25668#issuecomment-2473723592
